telluric/features.py

```diff
@@ -340,19 +353,34 @@ def reproject(self, *args, **kwargs):
     @classmethod
     def _from_record(cls, record, crs, schema=None):
         properties = cls._to_properties(record, schema)
-        raster = raster_from_assets(record.get("raster", {}))
-        return GeoFeatureWithRaster(raster, properties)
+        rasters = raster_from_assets(record.get("rasters", {}))
+        return GeoFeatureWithRaster(rasters, properties)
 
-    def copy_with(self, raster=None, properties=None):
+    def copy_with(self, rasters=None, properties=None):
         """Generate a new GeoFeatureWithRaster with different raster or preperties."""
-        if raster is None:
-            if self.raster.not_loaded():
-                raster = GeoRaster2.open(self.raster._filename)
-            else:
-                raster = self.raster.copy_with()
+        if rasters is None:
+            rasters = []
+            for raster in self.rasters:
+                if raster.not_loaded():
```

Comment:
maybe this if should be part of `copy_with` of the GeoRaster, and here you should call only `copy_with`

Reply:
yes